src/renderer/react/app/nodes/NodeManager.jsx

```diff
@@ -1,58 +1,79 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { compose } from 'recompose';
+import { compose, withStateHandlers } from 'recompose';
 import { connect } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
 
-import { nodeShape } from 'react/shapes/node';
-import { create, update, remove } from 'redux/stores/nodes/actions';
+import { create, remove, update } from 'redux/stores/nodes/actions';
 import isAuthenticated from 'react/hoc/isAuthenticated';
+import { nodeShape } from 'react/shapes/node';
+import Button from 'react/components/general/button/Button';
 
 import NodeManagerList from './nodeList/NodeManagerList';
-import PromptNewNodeModal from './editNodeModal/EditNodeModal';
+import EditNodeModal from './editNodeModal/EditNodeModal';
+import messages from './NodeManager.messages';
 
 const enhancer = compose(
   isAuthenticated,
+
   connect(
-    state => ({
-      nodeList: state.nodes,
-    }),
+    null,
     dispatch => ({
       onCreateNode: node => dispatch(create(node)),
       onUpdateNode: node => dispatch(update(node)),
-      onRemoveNode: node => dispatch(remove(node)),
+      onDeleteNode: node => dispatch(remove(node)),
     }),
   ),
+
+  withStateHandlers(
+    {
+      locked: true,
+      currentlySelectedNode: null,
+    },
+    {
+      toggleLock: ({ locked }) => () => ({ locked: !locked }),
+      selectNode: () => node => ({ currentlySelectedNode: node }),
+      unselectNode: () => () => ({ currentlySelectedNode: null }),
+    },
+  ),
 );
 
 class NodeManager extends React.PureComponent {
   static propTypes = {
-    /** List of nodes in the store. */
-    nodeList: PropTypes.objectOf(nodeShape).isRequired,
     /** On node creation. */
-    onCreateNode: PropTypes.func.isRequired,
+    onCreateNode: PropTypes.func,
     /** On node edition. */
-    onUpdateNode: PropTypes.func.isRequired,
+    onUpdateNode: PropTypes.func,
     /** On node deletion. */
-    onRemoveNode: PropTypes.func.isRequired,
+    onDeleteNode: PropTypes.func,
+
+    locked: PropTypes.bool.isRequired,
+    toggleLock: PropTypes.func.isRequired,
+    selectNode: PropTypes.func.isRequired,
+    unselectNode: PropTypes.func.isRequired,
+    currentlySelectedNode: PropTypes.shape(nodeShape),
   };
 
-  state = {
+  static defaultProps = {
+    onCreateNode: null,
+    onUpdateNode: null,
+    onDeleteNode: null,
     currentlySelectedNode: null,
   };
 
   /**
-   * Creates a new node, pass a parentId if it was given.
+   * Creates a new node.
    *
-   * @param {string} parentId - Id of parent.
+   * @param {object} node - Node that was clicked (which is going to become the parent).
    */
-  onAddNode = parentId => this.setState({ currentlySelectedNode: { parentId } });
+  onAddNode = node => this.props.selectNode({ parentId: node?.id || null });
```

Comment:
you can put your methods in

```js
withHandlers({
  onAddNode: ({ selectNode }) => node => selectNode({ parentId: node?.id || null }),
  // etc.
})
```
This way you can write it as a functional component.